streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/FunctionConversions.scala

```diff
@@ -34,6 +34,12 @@ import java.lang.{Iterable => JIterable}
  */
 object FunctionConversions {
 
+  implicit class ForeachActionFromFunction[K, V](val p: (K, V) => Unit) extends AnyVal {
+    def asForeachAction: ForeachAction[K, V] = new ForeachAction[K, V] {
+      override def apply(key: K, value: V): Unit = p(key, value)
+    }
+  }
+
```

Comment:
Technically, this is a public interface change, and would require a KIP.

Note that we don't really need the implicit; it just makes the internal implementation nicer. You could instead do:
```scala
def foreach(action: (K, V) => Unit): Unit =
    inner.foreach(new ForeachAction[K, V] {
      override def apply(key: K, value: V): Unit = action(key, value)
    })
```

right?

Reply:
Yes we could.